geopy/geocoders/geocodeapi.py

```diff
@@ -0,0 +1,235 @@
+
+        :param float circle_lon: Longitude for limit search to radius
+
+        :param int circle_radius: Radius around the coordinates circle_lat and circle_lon
+
+        :param float rect_min_lat: Minimum latitude for limit search to rectangle
+
+        :param float rect_min_lon: Minimum longitude for limit search to rectangle
+
+        :param float rect_max_lat: Maximum latitude for limit search to rectangle
+
+        :param float rect_max_lon: Maximum longitude for limit search to rectangle
+
+        :param float point_lat: Latitude for prioritize around a point
+
+        :param float point_lon: Longitude for prioritize around a point
+        """
+        params = self._get_params_geocode(
+            query,
+            size=size,
+            country=country,
+            circle_lat=circle_lat,
+            circle_lon=circle_lon,
+            circle_radius=circle_radius,
+            rect_min_lat=rect_min_lat,
+            rect_min_lon=rect_min_lon,
+            rect_max_lat=rect_max_lat,
+            rect_max_lon=rect_max_lon,
+            point_lon=point_lon,
+            point_lat=point_lat,
+        )
+
+        url = self.api_geocode + params
+
+        logger.debug('%s.geocode: %s', self.__class__.__name__, url)
+        callback = partial(self._parse_json, exactly_one=exactly_one)
+        return self._call_geocoder(url, callback, timeout=timeout, headers=self.headers)
+
+    def reverse(
+        self,
+        query,
+        *,
+        exactly_one=True,
+        timeout=DEFAULT_SENTINEL,
+        size=None,
+        layers=None,
+    ):
+        """
+        Return an address by location point.
+
+        :param query: The coordinates for which you wish to obtain the
+            closest human-readable addresses.
+        :type query: :class:`geopy.point.Point`, list or tuple of ``(latitude,
+            longitude)``, or string as ``"%(latitude)s, %(longitude)s"``.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+        :param int size: Limits the number of results to be returned
+
+        :param str layers: Filter search to a particular location type, e.g. venue,
+            address, street, country, region, locality
+
+        """
+        params = self._get_params_reverse(query, size=size, layers=layers)
+        url = '{}?{}'.format(self.api_reverse, params)
+        callback = partial(self._parse_json, exactly_one=exactly_one)
+        return self._call_geocoder(url, callback, timeout=timeout, headers=self.headers)
+
+    def _get_params_geocode(self, query, **kwargs):
+        main_param = '?text={}'.format(query)
```

Comment:
Passing user input without escaping is unsafe. Please remove that class-level `_param_templates` dict and switch to `urllib.parse.urlencode`. See other geocoders for example.